distributed/diagnostics/nvml.py

```diff
@@ -63,6 +63,24 @@ def _pynvml_handles():
     return pynvml.nvmlDeviceGetHandleByIndex(gpu_idx)
 
 
+def has_cuda_context():
+    """Check whether the current process already has a CUDA context created.
+
+    Returns
+    -------
+    ``False`` if current process has no CUDA context created, otherwise returns the
+    index of the device for which there's a CUDA context.
+    """
+    init_once()
```

Comment:
Should we be calling this in `has_cuda_context`? What happens if it gets called more than once?

Reply:
Yes, we should. We have to ensure (Py)NVML is initialized before calling it. The function itself takes care of the scenario where it's called multiple times: https://github.com/dask/distributed/blob/983914f83b3eceee212e3c44fd80494c8e11828d/distributed/diagnostics/nvml.py#L22-L23 .

Reply:
This makes more sense after our conversation below ( https://github.com/dask/distributed/pull/5308#discussion_r707579638 )